### PR TITLE
Sort targets by instance within a job

### PR DESCRIPTION
**Pull request by juliusv**
Fixes https://github.com/prometheus/prometheus/issues/2536

### CODE REVIEW

File: web/web.go [hunk 1]
```diff
@@ -379,6 +379,12 @@ func (h *Handler) targets(w http.ResponseWriter, r *http.Request) {
 		tps[job] = append(tps[job], t)
 	}
 
+	for _, targets := range tps {
+		sort.Slice(targets, func(i, j int) bool {
```
Comment (grobie) [inline web/web.go]: Using sort.Slice makes go1.8 a requirement now. If that's desired, we'll need to update the build+ci infrastructure first.

Comment (juliusv) [inline web/web.go]: Hm yeah, just noticed. I wonder what our policy around requiring up-to-date versions of Go is. I mean, most users won't build Prometheus themselves, so it should be ok?

I could implement an entire sort interface, but that's kind of ugly if this nice new way exists now :)

**Comment by grobie** [comment]
I think it's only a question of whether we want to release the next version
using the latest golang version. I think that depends on the timing: no if
the newest golang version is only a few days old, yes if it has matured and
no known bugs. For Prometheus v1.6.0 / go1.8 I think that's the case, so
updating the toolchain should be done.

**Comment by juliusv** [comment]
Ok, I'll send PRs for that.

**Comment by juliusv** [comment]
All green now - good to merge?

**Comment by grobie** [comment]
:+1:

**Pull request merged**